modules/unsupported/geojson-core/src/main/java/org/geotools/filter/expression/JSONNodePropertyAccessorFactory.java

```diff
@@ -0,0 +1,212 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2022, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.filter.expression;
+
+import com.fasterxml.jackson.core.JsonPointer;
+import com.fasterxml.jackson.databind.JsonNode;
+import java.time.Instant;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAccessor;
+import java.util.Date;
+import java.util.Map;
+import org.geotools.data.geojson.GeoJSONReader;
+import org.geotools.util.factory.Hints;
+import org.opengis.feature.Property;
+import org.opengis.feature.simple.SimpleFeature;
+
+/** Property Accessor Factory for {@link JsonNode} */
+public class JSONNodePropertyAccessorFactory implements PropertyAccessorFactory {
+
+    static PropertyAccessor JSONNODEPROPERTY = new JSONNodePropertyAccessor();
+    static final String ROOT = "/";
+
+    @Override
+    public PropertyAccessor createPropertyAccessor(
+            Class<?> type, String xpath, Class<?> target, Hints hints) {
+        if (xpath == null) {
+            return null;
+        }
+        return JSONNODEPROPERTY;
+    }
+
+    /** Property Accessor for {@link JsonNode} */
+    static class JSONNodePropertyAccessor implements PropertyAccessor {
+
+        private static final String JSON_NODE_DATE_FORMAT_OFF = "JSON_NODE_DATE_FORMAT_OFF";
+
+        @Override
+        public boolean canHandle(Object object, String xpath, Class target) {
+            try {
+                if (xpath != null && object != null && object instanceof SimpleFeature) {
+                    String[] parts = stripAndReturnHeadAndRest(xpath);
+                    SimpleFeature feature = (SimpleFeature) object;
+                    Property property = feature.getProperty(parts[0]);
```

Comment:
May be simpler if you called ``getAttribute`` instead (getting the value directly)

Reply:
changed